java-checks/src/main/java/org/sonar/java/checks/ServletInstanceFieldCheck.java

```diff
@@ -107,11 +109,19 @@ private void reportIssuesOnVariable() {
   private class AssignmentVisitor extends BaseTreeVisitor {
     @Override
     public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+      // handles e.g. "second = this.first * 2;" assignments -> no "this" prefix to member "second"
       if (tree.variable().is(Kind.IDENTIFIER)) {
```

Comment:
I have no strong opinion on that one, but since we're using `tree.variable()` four times in this method, maybe a local variable `var variable = tree.variable()` could be better? As said, no strong opinion here, because some people prefer _not_ to assign getter results to local variables.